src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -1726,8 +1714,83 @@ ValidPermissionedDomain::finalize(
         return true;
     };
 
-    return (sleStatus_[0] ? check(*sleStatus_[0], j) : true) &&
-        (sleStatus_[1] ? check(*sleStatus_[1], j) : true);
+    if (view.rules().enabled(fixPermissionedDomainInvariant))
+    {
+        // Nothing was changed by a failed tx. All good.
+        if (result != tesSUCCESS && sleStatus_.size() == 0)
+            return true;
+
```

Comment:
No permissioned domains should be affected if the transaction failed.

```suggestion
        // No permissioned domains should be affected if the transaction failed
        if (result != tesSUCCESS)
            // If nothing changed, all is good. If there were changes, that's bad.
            return sleStatus_.size() == 0;
```